pages/queryParamsTest.tsx

```diff
@@ -0,0 +1,35 @@
+import { useRouter } from 'next/router'
+import { useEffect, useState } from 'react'
+import { addURLQueryParam, removeURLQueryParam } from '../utils/queryParams'
+
+export default function Test() {
+  // IMPORTANT
+  // router.query isn't really updated until router.push() or router.replace() is called
+  // However, when called in the util functions, router.query is still outdated
+  // What happened was that hitting button 1 once would show that the path is /test
+  // but hitting it a SECOND time would then properly show that the path is /test?hi1=test
+  // This has to due with how nextjs router is made and timings/refreshing the page
+  // The workaround is to wait for the router object itself to update in a useEffect
+  // Here is how I would access the path in code -- I would just use a state variable
+  const router = useRouter()
```

Comment:
i am not sure about the feasibility of this, but could we create a custom hook for handling this? something like  `useRouterQuery`? that way we have less boilerplate in each component we want to use the utility function.

Reply:
That is a very good idea and something I didn't consider. 90e8301 should address this!